packages/dev-server-hmr/test/browser.test.ts

```diff
@@ -20,9 +20,17 @@ function trackErrors(page: Page) {
   return errors;
 }
 
-describe('browser tests', function () {
-  this.timeout(10000);
+type Predicate = (...args: unknown[]) => Promise<boolean> | boolean;
+
+async function pollFor(predicate: Predicate): Promise<void> {
```

Comment:
isn't it better to use Puppeteer API for that?

e.g. they have https://pptr.dev/api/puppeteer.page.waitforfunction
but in some cases other utils might be better

Reply:
Good shout, I'll look at that when I have time.